packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/DiagramPanel.tsx

```diff
@@ -44,6 +99,86 @@ export const DiagramPanel = ({
   const handleZoomOut = () => reactFlow.zoomOut({ duration: 200 });
   const handleShare = () => setState((prevState) => ({ ...prevState, dialogOpen: 'Share' }));
   const handleCloseDialog = () => setState((prevState) => ({ ...prevState, dialogOpen: null }));
+  const { editingContextId, diagramId } = useContext(DiagramContext);
```

Comment:
It would be better if you could extract all this hide/fade behavior out of the panel directly. An API like that would be better:

```
import { useFadeDiagramElements } from './fade/useFadeDiagramElements';
import { useHideDiagramElements } from './hide/useHideDiagramElements';

// ...

const { fadeDiagramElements } = useFadeDiagramElements();
const { hideDiagramElements } = useHideDiagramElements();

// ...

const onUnfadeAll = () => fadeDiagramElements([...getAllElementsIds()], false);
const onUnhideAll = () => hideDiagramElements([...getAllElementsIds()], false);
```

You don't need to give the functions `fadeDiagramElements` /  `hideDiagramElements` the `editingContextId` or the `representationId` since they can retrieve it from the `DiagramContext`. It would give us the ability to have the hide / fade specific code only at the given location.

You can also remove all hide/fade related code in the palette. Here a hook would be way better since we literally need to share the same code (useMutation, useMultiToast, etc) in multiple components

Reply:
I know that I am setting up an API with a slightly different name than the existing one (`useFadeDiagramElements` using the plural) but since the API allows for the change in visibility of multiple elements at once, it should be renamed eventually so let's prepare our code for that.

Reply:
By the way, you may not have to use a dedicated context here (contrary to the useDiagramDirectEdit) because you may not have a top level state that you need to access from anywhere in the application (while we have the `currentlyEditedLabelId` to access and manipulate from anywhere in the useDiagramDirectEdit hook)